### DISCUSSION TITLE
Removing SSP code from browser bundle

### DISCUSSION THREAD

**Discussion by ivansky** [vercel/next.js#29584] [Help] [answered]
Hi all!

I'm trying to understand the behaviour of removing server-side code from browser bundle, as I can see 
https://github.com/vercel/next.js/blob/5544adc481f8821567e947a6e6d51d9d68ebd367/packages/next/build/babel/plugins/next-ssg-transform.ts#L13-L17
in babel plugin there is remove  export name declaration.
https://github.com/vercel/next.js/blob/5544adc481f8821567e947a6e6d51d9d68ebd367/packages/next/build/babel/plugins/next-ssg-transform.ts#L248-L262

There is also a note about client-side bundle optimization 
https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering
> Note: You can import modules in top-level scope for use in getServerSideProps. Imports used in getServerSideProps will not be bundled for the client-side.
 
**The question:**
What if I have some sort of wrapper for ssp function? Will it be also removed due to tree shaking?
```ts
import { createPagePipe } from 'config/pageConfig'; // this module has a lot of server-specific code
export const getServerSideProps = createPagePipe({ ... });
```

**Comment by wwwdepot** [comment, marked as answer]
Next.js Team made this tool to preview code elimination https://next-code-elimination.vercel.app/

**Reply by ivansky** [reply]
For those who don't want to visit link above.
![Screenshot 2021-10-04 at 19 43 14](https://user-images.githubusercontent.com/3938372/135881924-ba296ec7-21a9-41f9-b454-4b51d723ba6b.png)
